sycl/include/CL/sycl/accessor.hpp

```diff
@@ -1600,29 +1600,62 @@ class accessor :
     return AccessorSubscript<Dims - 1>(*this, Index);
   }
 
+  // get_pointer()
+  // when dim==1, MData will have been preadjusted for faster access with []
+  // on device, getQualifiedPtr() returns MData, so we need to backjust it
+  // on host, getQualifiedPtr() does not return MData, no need to adjust
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
I would suggest to avoid this code duplication and have all 3 get_pointer specialization forward to the same one. That way the `__SYCL_DEVICE_ONLY__` and `(1 == AdjustedDim)` conditions are only inside that one function. If you anyhow planned to do that given it is still WIP please just ignore my comment.

Reply:
excellent suggestion. Done.